packages/db/src/models/granule.ts

```diff
@@ -20,15 +21,20 @@ export default class GranulePgModel extends BasePgModel<PostgresGranule, Postgre
     executionCumulusId: number,
     granulesExecutionsPgModel = new GranulesExecutionsPgModel()
   ) {
+    if (!granule.created_at) {
+      throw new Error(`To upsert granule record must have 'created_at' set: ${JSON.stringify(granule)}`);
+    }
     if (granule.status === 'running') {
       return knexOrTrx(this.tableName)
         .insert(granule)
-        .onConflict(['granule_id', 'collection_cumulus_id'])
+        .onConflict(['granule_id', 'collection_cumulus_id']) // should unique_constraints be a method on the db model?
         .merge({
           status: granule.status,
           timestamp: granule.timestamp,
           updated_at: granule.updated_at,
+          created_at: granule.created_at,
         })
+        .where(knexOrTrx.raw(`${this.tableName}.created_at <= to_timestamp(${translateDateToUTC(granule.created_at)})`))
```

Comment:
Thinking out loud: is there a minute loss of time precision when doing `date.getTime() / 1000`, so losing the last 3 digits of precision, but then you're comparing against something in the database which has those digits of precision naturally?

Reply:
Also, do we have to use a `raw` statement to use `to_timestamp`?

Reply:
Is `${this.tableName}` necessary in the raw statement if `created_at` only exists on the granules table in this query?

Reply:
>Is ${this.tableName} necessary in the raw statement if created_at only exists on the granules table in this query?

It appears to be - postgres was unhappy without the qualified reference.

Reply:
> Thinking out loud: is there a minute loss of time precision when doing date.getTime() / 1000, so losing the last 3 digits of precision, but then you're comparing against something in the database which has those digits of precision naturally?

Hmm, so 

to_timestamp:

 ( double precision ) → timestamp with time zone

Convert Unix epoch (seconds since 1970-01-01 00:00:00+00) to timestamp with time zone

and .getTime gets the time in ms (thus the conversion), however postgres uses ... microseconds.    I'm not sure node gives you a more precise time than ms without a refactor, more importantly I think *knex* isn't returning these values with more precision if you query the table (presumably as we're setting them via Date.now()): 

> foo[0]
{
  cumulus_id: '1',
  granule_id: '03d09',
  status: 'running',
  collection_cumulus_id: 1,
  created_at: 2021-03-09T08:40:34.072Z,
  updated_at: 2021-03-09T08:40:34.091Z,
  published: null,
  duration: null,
  time_to_archive: null,
  time_to_process: null,
  product_volume: null,
  error: null,
  cmr_link: null,
  pdr_cumulus_id: null,
  provider_cumulus_id: null,
  beginning_date_time: null,
  ending_date_time: null,
  last_update_date_time: null,
  processing_end_date_time: null,
  processing_start_date_time: null,
  production_date_time: null,
  query_fields: null,
  timestamp: null
}

also: 

> knex.raw('SELECT CURRENT_TIMESTAMP').then((f) => bar = f)
Promise { <pending> }
> bar
Result {
  command: 'SELECT',
  rowCount: 1,
  oid: null,
  rows: [ { current_timestamp: 2021-03-09T08:49:21.593Z } ],
  fields: [
    Field {
      name: 'current_timestamp',
      tableID: 0,
      columnID: 0,
      dataTypeID: 1184,
      dataTypeSize: 8,
      dataTypeModifier: -1,
      format: 'text'
    }
  ],
  _parsers: [ [Function: parseDate] ],
  _types: TypeOverrides {
    _types: {
      getTypeParser: [Function: getTypeParser],
      setTypeParser: [Function: setTypeParser],
      arrayParser: [Object],
      builtins: [Object]
    },
    text: {},
    binary: {}
  },
  RowCtor: null,
  rowAsArray: false
}
> bar.Result.rows
Uncaught TypeError: Cannot read property 'rows' of undefined
> bar.rows
[ { current_timestamp: 2021-03-09T08:49:21.593Z } ]
> bar.rows[0]
{ current_timestamp: 2021-03-09T08:49:21.593Z }
> bar.rows[0].current_timestamp
2021-03-09T08:49:21.593Z


There's a valid question of if we *want* more precision in those timestamps, but at least in the context of this effort, we likely have a real problem if microseconds come into play re: workflow start times.

Reply:
That all seems fine. My concern, which I'm not even sure make sense, was just whether comparing against `translateDateToUTC(granule.created_at)` could give unexpected results, but it seems like the answer is no